### ISSUE TITLE
Add import() support

### ISSUE THREAD

**Issue by chicoxyzzy** [cramforce/splittable#32]
It should replace `System.import` method
https://github.com/tc39/proposal-dynamic-import

The proposal can possibly [move to stage 3 next week](https://github.com/tc39/agendas/blob/master/2016/11.md)

**Comment by cramforce** [comment]
Happy to take a PR :)

**Comment by chicoxyzzy** [comment]
It is on stage 3 now

**Comment by chicoxyzzy** [comment]
`import` is reserved words so dynamic imports can't be polyfilled

**Comment by chicoxyzzy** [comment]
There is [https://github.com/babel/babel/tree/master/packages/babel-plugin-syntax-dynamic-import](babel-plugin-syntax-dynamic-import) so I suppose we can use custom plugin to transpile it to current `System.import`. @cramforce are you ok with that?

**Comment by cramforce** [comment]
Definitely would need to be polyfilled using rewriting of files. The main
thing is resolving the module identifier to a bundle URL.

**Comment by cramforce** [comment]
Just saw the second comment. My email was delayed.

Yeah, you can insert that path in here: https://github.com/cramforce/splittable/blob/master/splittable.js#L223

And then look for 1System.import` in the evil regex below and that might work!

**Comment by chicoxyzzy** [comment]
I just created [babel plugin](https://github.com/chicoxyzzy/babel-plugin-dynamic-import-system-import), I think it could be added to line you mentioned but I tests look complicated. I didn't figure out how to add tests for `import()`

**Comment by cramforce** [comment]
Cool. I can take it from there. Does the plugin do nothing except `s/import/System.import`. One very useful feature would be a hook to control rewriting of the module argument (Going from module identifier to URL).

**Comment by chicoxyzzy** [comment]
Yes, it replaces dynamic `import`s with `System.import`s but not just `s/import/System.import` because we should replace only import calls, not static imports.

> One very useful feature would be a hook to control rewriting of the module argument (Going from module identifier to URL).

Not sure if this logic should be in babel plugin. `System.import` needs this too, right?

**Comment by cramforce** [comment]
It cannot be correctly done at runtime. The nice thing about import is that it resolves a relative module name. A regular function call cannot know what `'../foo'` means, because it doesn't know about filenames anymore.

**Comment by chicoxyzzy** [comment]
That's true. Can you explain what changes do you want from transformer?

**Comment by cramforce** [comment]
@chicoxyzzy It would either be configurable or something predictable. E.g. if the resulting path was always a relative path from the project root instead of the current file, then everything else could be done at runtime.

**Comment by chicoxyzzy** [comment]
ok I'll try to add an option for import specifier to be exactly `StringLiteral` and be a relative path.
